sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -603,6 +613,175 @@ describe("StudentAppealStudentsController(e2e)-submitStudentAppeal", () => {
     );
   });
 
+  it(
+    "Should create room and board costs appeal " +
+      "when student submit an appeal for a program year which is eligible for appeal process.",
+    async () => {
+      // Arrange
+      // Create student to submit application.
+      const student = await saveFakeStudent(appDataSource);
+      // Create application submit appeal with eligible program year.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          programYear: recentActiveProgramYear,
+        },
+        { applicationStatus: ApplicationStatus.Completed },
+      );
+      // Create a temporary file for room and board costs appeal.
+      const roomAndBoardFile = await saveFakeStudentFileUpload(
+        appDataSource,
+        {
+          student,
+          creator: student.user,
+        },
+        { fileOrigin: FileOriginType.Temporary },
+      );
+      // Prepare the data to request a change of financial information.
+      const roomAndBoardAppealData = {
+        roomAndBoardAmount: 561,
+        roomAndBoardSituations: {
+          parentUnEmployed: false,
+          parentEarnLowIncome: false,
+          parentReceiveIncomeAssistance: false,
+          livingAtHomePayingRoomAndBoard: true,
+          parentReceiveCanadaPensionOrOldAgeSupplement: false,
+        },
+        roomAndBoardSupportingDocuments: [
+          {
+            url: `student/files/${roomAndBoardFile.uniqueFileName}`,
+            hash: "",
+            name: roomAndBoardFile.uniqueFileName,
+            size: 4,
+            type: "text/plain",
+            storage: "url",
+            originalName: roomAndBoardFile.fileName,
+          },
+        ],
+      };
+      const payload: StudentAppealAPIInDTO = {
+        studentAppealRequests: [
+          {
+            formName: ROOM_AND_BOARD_COSTS_FORM_NAME,
+            formData: roomAndBoardAppealData,
```

Comment:
For my understanding, was there a need to mock the `formData` for this E2E?

Reply:
I got that some sort of formData was required, but the question was related to how close to the real-world payload it needs to be.
Not a blocker, but I would avoid to avoid setting it as the standard unless we have the form.io complete E2E integration in place.